ftmstore/dataset.py

```diff
@@ -165,8 +167,11 @@ def __iter__(self):
         return self.iterate()
 
     def __len__(self):
-        q = select([func.count(distinct(self.table.c.id))])
-        return self.store.engine.execute(q).scalar()
+        q = select(func.count(distinct(self.table.c.id)))
+        with self.store.engine.connect() as conn:
+            res = conn.execute(q)
```

Comment:
The `execute()` method is now called on a context manager object which is a connection (`conn`).  The `execute()` method was previously called on the `engine` object, which is no longer possible in SQLAlchemy 2.0. It's an interface change.

This part in my Pull Request description references this change: [Many Choices becomes One Choice](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#many-choices-becomes-one-choice) - call execute() on a connection, not on the Engine object

Reply:
Yes, I read that! :) Just wanted to clarify that this is indeed just an API change and doesn't change the behavior. (For example, I was wondering if the previous 1.4 API did some magic to prevent having to open/close a connection for every statement), but I guess they would've mentioned that in the migration guide.